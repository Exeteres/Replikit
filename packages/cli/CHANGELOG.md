# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.9.0](https://github.com/Exeteres/Replikit/compare/v0.8.0...v0.9.0) (2020-06-28)

**Note:** Version bump only for package @replikit/cli





# [0.8.0](https://github.com/Exeteres/Replikit/compare/v0.7.2...v0.8.0) (2020-06-27)

**Note:** Version bump only for package @replikit/cli





## [0.7.2](https://github.com/Exeteres/Replikit/compare/v0.7.1...v0.7.2) (2020-06-23)

**Note:** Version bump only for package @replikit/cli





# [0.7.0](https://github.com/Exeteres/Replikit/compare/v0.6.0...v0.7.0) (2020-06-23)


### Bug Fixes

* **cli:** Fix generated package.json and tsconfig ([83854b5](https://github.com/Exeteres/Replikit/commit/83854b53c86f691613f42218a8f321270ad22f93))





# [0.6.0](https://github.com/Exeteres/Replikit/compare/v0.5.3...v0.6.0) (2020-06-21)

**Note:** Version bump only for package @replikit/cli





## [0.5.3](https://github.com/Exeteres/Replikit/compare/v0.5.2...v0.5.3) (2020-06-20)

**Note:** Version bump only for package @replikit/cli





## [0.5.2](https://github.com/Exeteres/Replikit/compare/v0.5.1...v0.5.2) (2020-06-20)

**Note:** Version bump only for package @replikit/cli





# 0.5.0 (2020-06-20)


### Bug Fixes

* **cli:** Add @types/node to generated package.json ([4787912](https://github.com/Exeteres/Replikit/commit/4787912620fd2e332728a687ae92f5563fd6f350))
* **cli:** Add missing dependencies ([a5ba2c1](https://github.com/Exeteres/Replikit/commit/a5ba2c1f9ba83692b118655b519d58b14d85f76f))
* **cli:** Fix "workspaces" section in generated package.json ([3dc4214](https://github.com/Exeteres/Replikit/commit/3dc4214a8f929afca7015d9cd3942276ec779c41))
* **cli:** Fix configuration override ([1866a8a](https://github.com/Exeteres/Replikit/commit/1866a8a005055d4a375104964c3d01508ac56cef))


### Features

* **cli:** Add build command ([5c9dc57](https://github.com/Exeteres/Replikit/commit/5c9dc57b9ac2428f4970096dcfc6ae75b2e4dd27))
* **cli:** Add scripts section and replikit as dev dependency to generated package.json ([2f5dd6f](https://github.com/Exeteres/Replikit/commit/2f5dd6f35f47f5fa50109b7155f56109b1ba3efc))






# [0.4.0](https://github.com/Exeteres/Replikit/compare/v0.3.0...v0.4.0) (2020-06-19)


### Features

* **cli:** Add build command ([0212d7f](https://github.com/Exeteres/Replikit/commit/0212d7f9c87d517b8cf220be9b32376551c3603b))





# [0.3.0](https://github.com/Exeteres/Replikit/compare/v0.2.0...v0.3.0) (2020-06-14)


### Bug Fixes

* **cli:** Add @types/node to generated package.json ([3d4fc2b](https://github.com/Exeteres/Replikit/commit/3d4fc2b1c5d0daad6040fe956d3cf164f3b1a227))
* **cli:** Fix configuration override ([d70cddb](https://github.com/Exeteres/Replikit/commit/d70cddb919469e90ed29a1bead411908bf370b57))





# [0.2.0](https://github.com/Exeteres/Replikit/compare/v0.1.0...v0.2.0) (2020-06-13)


### Bug Fixes

* **cli:** Add missing dependencies ([a655059](https://github.com/Exeteres/Replikit/commit/a65505938400bf29c2b890966a2685ac7618d989))
* **cli:** Fix "workspaces" section in generated package.json ([12aca8b](https://github.com/Exeteres/Replikit/commit/12aca8b5f729834c270abfed5606e932b987d48b))


### Features

* **cli:** Add scripts section and replikit as dev dependency to generated package.json ([64399a2](https://github.com/Exeteres/Replikit/commit/64399a242938b1e12baef40a75cf0d6cfeec3b0f))
