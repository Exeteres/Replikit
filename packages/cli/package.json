{
    "name": "@replikit/cli",
    "version": "0.5.3",
    "license": "MIT",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "typings"
    ],
    "scripts": {
        "build": "ttsc -p tsconfig.build.json",
        "dev": "ts-node -r tsconfig-paths/register src/main.ts"
    },
    "bin": {
        "replikit": "./dist/main.js"
    },
    "dependencies": {
        "@replikit/chalk": "^0.5.3",
        "@replikit/core": "^0.5.3",
        "commander": "^5.1.0",
        "dotenv": "^8.2.0",
        "execa": "^4.0.0",
        "fs-extra": "^9.0.0",
        "inquirer": "^7.1.0",
        "native-addon-loader": "^2.0.1",
        "require-from-string": "^2.0.2",
        "ts-loader": "^7.0.5",
        "ts-morph": "^7.0.3",
        "ts-node": "^8.0.2",
        "ts-node-dev": "^1.0.0-pre.44",
        "tsconfig-paths": "^3.9.0",
        "ttypescript": "^1.5.10",
        "typescript": "^3.8.3",
        "webpack": "^4.43.0",
        "webpack-virtual-modules": "^0.2.2",
        "webpackbar": "^4.0.0"
    },
    "devDependencies": {
        "@types/fs-extra": "^8.1.0",
        "@types/inquirer": "^6.5.0",
        "@types/require-from-string": "^1.2.0",
        "@types/webpack": "^4.41.17",
        "@types/webpack-virtual-modules": "^0.1.0",
        "@types/webpackbar": "^4.0.0"
    }
}
