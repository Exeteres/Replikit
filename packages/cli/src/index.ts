/// <reference path="../typings/index.d.ts" />

export * from "./controllers/shellController";
export * from "./controllers/gitController";
export * from "./controllers/pm/pmType";
export * from "./controllers/pm/pmController";
export * from "./controllers/pm/npmController";
export * from "./controllers/pm/yarnController";
export * from "./controllers/pm/pmDetector";

export * from "./project/moduleManager";
export * from "./project/projectManager";
export * from "./project/configManager";

export * from "./actions/initProject";
export * from "./actions/createModule";

export * from "./errors";
export * from "./utils";
export * from "./startup";
export * from "./webpack";

export * from "./shared";

import "./args";

import "./commands/init";
import "./commands/createModule";
import "./commands/dev";
import "./commands/build";
import "./commands/addExternal";
