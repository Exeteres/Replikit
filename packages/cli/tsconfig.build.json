{
    "extends": "../../tsconfig.build.json",
    "compilerOptions": {
        "outDir": "dist",
        "paths": {
            "@replikit/cli": ["cli/src"]
        }
    },
    "include": ["src"]
}
