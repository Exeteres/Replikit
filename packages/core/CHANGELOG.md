# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.17.2](https://github.com/Exeteres/Replikit/compare/v0.17.1...v0.17.2) (2020-08-24)

**Note:** Version bump only for package @replikit/core





## [0.17.1](https://github.com/Exeteres/Replikit/compare/v0.17.0...v0.17.1) (2020-08-21)

**Note:** Version bump only for package @replikit/core





# [0.17.0](https://github.com/Exeteres/Replikit/compare/v0.16.0...v0.17.0) (2020-08-21)


### Features

* **core:** Add Identifier type ([e97ef90](https://github.com/Exeteres/Replikit/commit/e97ef90d24068fe4c16842b17bae0b8e6570ec47))





# [0.16.0](https://github.com/Exeteres/Replikit/compare/v0.15.1...v0.16.0) (2020-08-18)


### Bug Fixes

* **core:** Fix bug in text tokenization system ([43404bf](https://github.com/Exeteres/Replikit/commit/43404bffdb878e1f4abd18f69970cd6520251dd9))


### Features

* **core:** Add support for message headers ([5754c94](https://github.com/Exeteres/Replikit/commit/5754c94e0f89edb4ff3dde4fe410ade303bfb58f))





## [0.15.1](https://github.com/Exeteres/Replikit/compare/v0.15.0...v0.15.1) (2020-08-13)

**Note:** Version bump only for package @replikit/core





# [0.15.0](https://github.com/Exeteres/Replikit/compare/v0.14.0...v0.15.0) (2020-08-13)

**Note:** Version bump only for package @replikit/core





# [0.14.0](https://github.com/Exeteres/Replikit/compare/v0.13.0...v0.14.0) (2020-08-05)


### Bug Fixes

* **cli:** Fix typings ([9851fe5](https://github.com/Exeteres/Replikit/commit/9851fe5b87e07318da5c085f4b1bbddb23984136))





# [0.13.0](https://github.com/Exeteres/Replikit/compare/v0.12.1...v0.13.0) (2020-08-04)


### Bug Fixes

* **core:** Fix constructor type ([9687c48](https://github.com/Exeteres/Replikit/commit/9687c48773917919a2d8c38add27fba15cc11f00))


### Features

* **core:** Add ability to use pipe method in builders ([95c3cca](https://github.com/Exeteres/Replikit/commit/95c3cca0d1f4c5203fde3386c3775ecc77b92a1b))





# [0.12.0](https://github.com/Exeteres/Replikit/compare/v0.11.2...v0.12.0) (2020-07-30)


### Bug Fixes

* **core:** Rename query field to text ([4b0a930](https://github.com/Exeteres/Replikit/commit/4b0a930a4a840efcf56bdc6dfcf0b77a9430a30c))


### Features

* **core:** Refactor types and add support for inline queries ([2e52258](https://github.com/Exeteres/Replikit/commit/2e52258587cf0f2d31061c991eb2489ec7acda9e))





## [0.11.2](https://github.com/Exeteres/Replikit/compare/v0.11.1...v0.11.2) (2020-07-24)

**Note:** Version bump only for package @replikit/core





## [0.11.1](https://github.com/Exeteres/Replikit/compare/v0.11.0...v0.11.1) (2020-07-24)

**Note:** Version bump only for package @replikit/core






# [0.11.0](https://github.com/Exeteres/Replikit/compare/v0.10.1...v0.11.0) (2020-07-24)


### Bug Fixes

* **core:** Fix deepmerge ([132b184](https://github.com/Exeteres/Replikit/commit/132b18493e02603c195af70a2cfff0e9c31dc383))


### Features

* **sessions:** Add sessions package ([e7a0adf](https://github.com/Exeteres/Replikit/commit/e7a0adf3bbe35091bad6dcf28395e3c389a20a83))





## [0.10.1](https://github.com/Exeteres/Replikit/compare/v0.10.0...v0.10.1) (2020-06-30)

**Note:** Version bump only for package @replikit/core






# [0.10.0](https://github.com/Exeteres/Replikit/compare/v0.9.0...v0.10.0) (2020-06-29)

**Note:** Version bump only for package @replikit/core





# [0.9.0](https://github.com/Exeteres/Replikit/compare/v0.8.0...v0.9.0) (2020-06-28)


### Features

* Add advanced metadata processing ([54c7820](https://github.com/Exeteres/Replikit/commit/54c782021f5ebc16d784f210d273f8d7490d2af5))





# [0.8.0](https://github.com/Exeteres/Replikit/compare/v0.7.2...v0.8.0) (2020-06-27)


### Features

* **core:** Add core support for replies ([21f3318](https://github.com/Exeteres/Replikit/commit/21f3318f7bd031233856e7a1e1ddd29d73ef5f04))





## [0.7.2](https://github.com/Exeteres/Replikit/compare/v0.7.1...v0.7.2) (2020-06-23)

**Note:** Version bump only for package @replikit/core





# [0.7.0](https://github.com/Exeteres/Replikit/compare/v0.6.0...v0.7.0) (2020-06-23)

**Note:** Version bump only for package @replikit/core





# [0.6.0](https://github.com/Exeteres/Replikit/compare/v0.5.3...v0.6.0) (2020-06-21)

**Note:** Version bump only for package @replikit/core





## [0.5.3](https://github.com/Exeteres/Replikit/compare/v0.5.2...v0.5.3) (2020-06-20)

**Note:** Version bump only for package @replikit/core





## [0.5.2](https://github.com/Exeteres/Replikit/compare/v0.5.1...v0.5.2) (2020-06-20)

**Note:** Version bump only for package @replikit/core





# 0.5.0 (2020-06-20)


### Features

* **cli:** Add build command ([5c9dc57](https://github.com/Exeteres/Replikit/commit/5c9dc57b9ac2428f4970096dcfc6ae75b2e4dd27))






# [0.4.0](https://github.com/Exeteres/Replikit/compare/v0.3.0...v0.4.0) (2020-06-19)


### Features

* **cli:** Add build command ([0212d7f](https://github.com/Exeteres/Replikit/commit/0212d7f9c87d517b8cf220be9b32376551c3603b))





# [0.3.0](https://github.com/Exeteres/Replikit/compare/v0.2.0...v0.3.0) (2020-06-14)

**Note:** Version bump only for package @replikit/core





# [0.2.0](https://github.com/Exeteres/Replikit/compare/v0.1.0...v0.2.0) (2020-06-13)

**Note:** Version bump only for package @replikit/core
