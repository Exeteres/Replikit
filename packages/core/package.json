{
    "name": "@replikit/core",
    "version": "0.11.1",
    "license": "MIT",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "typings"
    ],
    "scripts": {
        "build": "tsc -p tsconfig.build.json"
    }
}
