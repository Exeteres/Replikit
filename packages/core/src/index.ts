/// <reference path="../typings/index.d.ts" />

export * from "./models/channelType";
export * from "./models/attachmentType";
export * from "./models/textTokenKind";
export * from "./models/textTokenProp";
export * from "./models/logLevel";

export * from "./composition/compositionItem";
export * from "./composition/compositionFactory";
export * from "./composition/compositionContext";
export * from "./composition/composition";

export * from "./errors";
export * from "./utils";
export * from "./hookStorage";
export * from "./config";
export * from "./eventHandler";
export * from "./cacheManager";
export * from "./textTokenizer";
export * from "./textFormatter";
export * from "./controller";
export * from "./controllerStorage";

export * from "./logging/consoleLogProvider";
export * from "./logging/logManager";
export * from "./logging/logger";

import "./startup";
