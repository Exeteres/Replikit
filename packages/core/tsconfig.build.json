{
    "extends": "../../tsconfig.build.json",
    "compilerOptions": {
        "outDir": "dist",
        "paths": {
            "@replikit/core": ["core/src"]
        }
    },
    "include": ["src"]
}
