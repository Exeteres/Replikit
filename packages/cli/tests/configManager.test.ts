import { code } from "@replikit/test-utils";
import { ConfigManager } from "@replikit/cli";

describe("TSConfigManager", () => {
    it("should add an import declaration to file without imports", () => {
        const config = code`
        import { Configuration } from "@replikit/core";

        const config: Configuration = {};

        export default config;`;

        const manager = new ConfigManager();
        manager.load(config);
        manager.addModule("@replikit/test");

        const result = manager.serialize();
        expect(result).toMatchSnapshot();
    });

    it("should add an import declaration to file with one import", () => {
        const config = code`
        import { Configuration } from "@replikit/core";

        import "@replikit/another";

        const config: Configuration = {};

        export default config;`;

        const manager = new ConfigManager();
        manager.load(config);
        manager.addModule("@replikit/test");

        const result = manager.serialize();
        expect(result).toMatchSnapshot();
    });

    it("should remove the import declaration from file with one import", () => {
        const config = code`
        import { Configuration } from "@replikit/core";

        import "@replikit/test";

        const config: Configuration = {};

        export default config;`;

        const manager = new ConfigManager();
        manager.load(config);
        manager.removeModule("@replikit/test");

        const result = manager.serialize();
        expect(result).toMatchSnapshot();
    });

    it("should remove the import declaration from typescript file with multiple imports", () => {
        const config = code`
        import { Configuration } from "@replikit/core";

        import "@replikit/another";
        import "@replikit/test";

        const config: Configuration = {};

        export default config;`;

        const manager = new ConfigManager();
        manager.load(config);
        manager.removeModule("@replikit/test");

        const result = manager.serialize();
        expect(result).toMatchSnapshot();
    });

    it("should check the typescript import declaration existence", () => {
        const config = code`
        import { Configuration } from "@replikit/core";

        import "@replikit/test";

        const config: Configuration = {};

        export default config;`;

        const manager = new ConfigManager();
        manager.load(config);

        expect(manager.checkModule("@replikit/test")).toBeTruthy();
        expect(manager.checkModule("@replikit/another")).toBeFalsy();
    });

    it("should init new configuration", () => {
        const manager = new ConfigManager();
        manager.init();

        const result = manager.serialize();
        expect(result).toMatchSnapshot();
    });

    it("should compile to an empty importless js configuration", () => {
        const manager = new ConfigManager();
        manager.init();

        const result = manager.compile();
        expect(result).toMatchSnapshot();
    });

    it("should compile to an importless js configuration with modules", () => {
        const config = code`
        import { Configuration } from "@replikit/core";

        import "@replikit/test";

        const config: Configuration = {
            test: { test: "test" }
        };

        export default config;`;

        const manager = new ConfigManager();
        manager.load(config);

        const result = manager.compile();
        expect(result).toMatchSnapshot();
    });
});
