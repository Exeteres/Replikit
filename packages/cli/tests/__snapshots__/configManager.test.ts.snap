// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TSConfigManager should add an import declaration to file with one import 1`] = `
"import { Configuration } from \\"@replikit/core\\";

import \\"@replikit/another\\";
import \\"@replikit/test\\";

const config: Configuration = {};

export default config;
"
`;

exports[`TSConfigManager should add an import declaration to file without imports 1`] = `
"import { Configuration } from \\"@replikit/core\\";

import \\"@replikit/test\\";

const config: Configuration = {};

export default config;
"
`;

exports[`TSConfigManager should compile to an empty importless js configuration 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
var config = {};
exports.default = config;
exports.modules = [];"
`;

exports[`TSConfigManager should compile to an importless js configuration with modules 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
var config = {
    test: { test: \\"test\\" }
};
exports.default = config;
exports.modules = [\\"@replikit/test\\"];"
`;

exports[`TSConfigManager should init new configuration 1`] = `
"import { Configuration } from \\"@replikit/core/typings\\";

const config: Configuration = {};

export default config;
"
`;

exports[`TSConfigManager should remove the import declaration from file with one import 1`] = `
"import { Configuration } from \\"@replikit/core\\";

const config: Configuration = {};

export default config;
"
`;

exports[`TSConfigManager should remove the import declaration from typescript file with multiple imports 1`] = `
"import { Configuration } from \\"@replikit/core\\";

import \\"@replikit/another\\";

const config: Configuration = {};

export default config;
"
`;
